addons/sale_mrp/models/sale_order_line.py

```diff
@@ -120,8 +120,9 @@ def _get_incoming_outgoing_moves_filter(self):
         :rtype: dict
         """
         return {
-            'incoming_moves': lambda m: m.location_dest_id.usage == 'customer' and
-                        (not m.origin_returned_move_id or (m.origin_returned_move_id and m.to_refund)),
+            'incoming_moves': lambda m: (m.location_dest_id.usage == 'customer' and
+                        (not m.origin_returned_move_id or (m.origin_returned_move_id and m.to_refund)))
+                        or (not m.picked and m.location_final_id.usage == 'customer'),
```

Comment:
@Whenrow starting V17.2 after confirming the SO in the case of multi-step delivery
not all pickings are created for example (pick - ship) only the first picking is created and the second is only created after the first one is validated
when checking the quantity being delivered there's no move going to the customer at this moment (before we validate the first picking)

Reply:
I would leave like this for now. this is the new push rule mechanism and we changed the delivery route but the pull rules still work like before.

Reply:
> I would leave like this

you mean the way I did it ? because right now updating the quantity the SO line will for example from 30 to 70
would lead 70 more being added to the move instead of just 40 for the difference 

since no move to the client exist it doesn't see that there already is a procurement for 30

Reply:
Ok I see the issue. However, I would do the same condition as for classic stock moves, not checking the `picked` value.

https://github.com/odoo/odoo/blob/171b910282765454442fdd1aea3c50d8434ee128/addons/sale_stock/models/sale_order_line.py#L270-L276

Reply:
> Ok I see the issue. However, I would do the same condition as for classic stock moves, not checking the `picked` value.
> 
> https://github.com/odoo/odoo/blob/171b910282765454442fdd1aea3c50d8434ee128/addons/sale_stock/models/sale_order_line.py#L270-L276

wouldn't this pass all the moves I actually just want to filter out the last one in the chain to get its quantity

Reply:
I was wrong we actually want the first move of the chain (rule in triggering_rule_ids) in the case of computing the procurement qty